boards/x86/acrn/doc/index.rst

```diff
@@ -0,0 +1,123 @@
+.. _acrn:
+
+ACRN UOS (Guest Operating System)
+#################################
+
+Overview
+********
+
+This board configuration defines an ACRN User OS execution environment for
+running Zephyr RTOS applications.
+
+ACRN is a flexible, lightweight reference hypervisor, built with real-time
+and safety-criticality in mind, optimized to streamline embedded development
+through an open source platform. Check out the `Introduction to Project ACRN
+<https://projectacrn.github.io/latest/introduction/>`_ for more information.
+
+This baseline configuration can be used as a starting point for creating
+custom ACRN UOS configurations. It currently supports the following devices:
+
+* I/O APIC
+* local APIC timer
+* NS16550 UARTs
+
+Serial Ports
+------------
+
+The serial ports are assumed present at traditional ``COM1:`` and ``COM2:``
+I/O-space addresses (based at ``0x3f8`` and ``0x2f8``, respectively). Only
+polled operation is supported in this baseline configuration, as IRQ
+assignments under ACRN are configurable (and frequently non-standard).
+Interrupt-driven and MMIO operation are also possible.
+
+Building and Running
+********************
+
+#. The build process for ACRN targets is similar to other boards.  For example,
+you can build the :ref:`hello_world` sample for ACRN with:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: acrn
+   :goals: build
+
+#. Build grub2 boot loader image
+
+   We can build grub2 bootloader for zephyr using
+   ``boards/x86/common/scripts/build_grub.sh``
+   which locate in `Zephyr Sourcecode
+   <https://github.com/zephyrproject-rtos/zephyr>`_.
+
+   .. code-block:: none
+
+      $ ./boards/x86/common/scripts/build_grub.sh x86_64
+
+   Find the binary at boards/x86/common/scripts/grub/bin/grub_x86_64.efi.
+
+#. Preparing the boot device
+
+   .. code-block:: none
+
+      $ dd if=/dev/zero of=zephyr.img bs=1M count=1024
+      $ mkfs.vfat -F 32 zephyr.img
+      $ losetup -f -P --show zephyr.img
+      $ mount /dev/loop0 /mnt
+
+   Create the following directories.
```

Comment:
I'd rather see the actual commands that are used (use a .. code-block) to do these actions instead of a description of what needs to be done.

```
.. code-block:: none

  $ mkdir -p /mnt/efi/boot
  $ cp grub_x86_64.efi /mnt/efi/boot

  $ mkdir -p /mnt/kernel 
  $ cp zephyr.elf /mnt/kernel


  ... etc.
```

Reply:
Actually, better to use this:
```
sudo cp boards/x86/common/scripts/grub/bin/grub_x86_64.efi /mnt/efi/boot/bootx64.efi
```
Using `bootx64.efi` as the target name will make it so that the `OVMF.fd` firmware will automatically find and load it (so it saves us typing stuff on the EFI shell)

Reply:
done